### ISSUE TITLE
errors on design-system components about explicit file extensions

### ISSUE THREAD

**Issue by tunnckoCore** [haydenbleasel/next-forge#277]
Latest `next-forge`, init-ed before an hour or so. Not touched any typescript configs.

![image](https://github.com/user-attachments/assets/3de7581e-9f52-4636-bcc5-38b1c3fbeafc)

> ts: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './components/ui/sonner.js'?

I'm fan of the explicit, but this might not work ok for Shad and/or Nextjs, don't know. 

base tsconfig

```json
{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Default",
  "compilerOptions": {
    "declaration": true,
    "declarationMap": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "incremental": false,
    "isolatedModules": true,
    "lib": ["es2022", "DOM", "DOM.Iterable"],
    "module": "NodeNext",
    "moduleDetection": "force",
    "moduleResolution": "NodeNext",
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "strict": true,
    "target": "ES2022",
    "strictNullChecks": true,
    "noImplicitAny": false,
    "noEmit": true,
    "allowImportingTsExtensions": true
  }
}

```

tsconfig inside `design-system`

```json
{
  "extends": "@repo/typescript-config/react-library.json",
  "compilerOptions": {
    "baseUrl": ".",
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@repo/*": ["../*"],
      "@repo/design-system/*": ["./*"]
    }
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"]
}

```

react-lib tsconfig

```json
{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "React Library",
  "extends": "./base.json",
  "compilerOptions": {
    "jsx": "react-jsx"
  }
}

```

**Comment by haydenbleasel** [comment]
Hey @tunnckoCore! Thanks for reporting this 😊

I took a deep dive into the TypeScript configs and I believe this is a configuration mismatch rather than a bug in next-forge itself.

### What's happening

The error `Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'` occurs because the design-system package is inheriting `moduleResolution: "NodeNext"` from the base tsconfig. With NodeNext, TypeScript strictly requires explicit file extensions on relative imports (e.g. `./components/ui/sonner.js` instead of `./components/ui/sonner`).

### Your config vs. the expected config

In your tsconfig snippets, the design-system extends `react-library.json`:

```json
{
  "extends": "@repo/typescript-config/react-library.json"
}
```

`react-library.json` only adds `"jsx": "react-jsx"` on top of `base.json`, which means it inherits `"moduleResolution": "NodeNext"` — and that's what triggers the error.

The design-system should instead extend `nextjs.json`:

```json
{
  "extends": "@repo/typescript-config/nextjs.json"
}
```

`nextjs.json` overrides the base config with:
- `"module": "ESNext"`
- `"moduleResolution": "Bundler"`

The `Bundler` module resolution strategy doesn't require explicit file extensions on relative imports, so `from "./components/ui/sonner"` works without issue.

### TL;DR

The design-system's tsconfig should extend `@repo/typescript-config/nextjs.json` (which uses `moduleResolution: "Bundler"`), not `react-library.json` (which inherits `moduleResolution: "NodeNext"`). This is how the current version of next-forge is set up, so if you re-init you should be good to go!
